src/microbe_stage/editor/OrganellePopupMenu.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System;
+using System.Globalization;
+using Godot;
+
+/// <summary>
+///   Manages a custom context menu solely for showing list of options for a placed organelle
+///   in the microbe editor.
+/// </summary>
+public class OrganellePopupMenu : PopupPanel
+{
+    [Export]
+    public NodePath SelectedOrganelleNameLabelPath;
+
+    [Export]
+    public NodePath DeleteButtonPath;
+
+    [Export]
+    public NodePath MoveButtonPath;
+
+    private Label selectedOrganelleNameLabel;
+    private Button deleteButton;
+    private Button moveButton;
+
+    private bool showPopup;
+    private OrganelleTemplate selectedOrganelle;
+    private bool enableDelete = true;
+    private bool enableMove = true;
+
+    [Signal]
+    public delegate void DeletePressed();
+
+    [Signal]
+    public delegate void MovePressed();
+
+    public bool ShowPopup
+    {
+        get => showPopup;
+        set
+        {
+            showPopup = value;
+
+            if (ShowPopup)
+            {
+                RectPosition = GetViewport().GetMousePosition();
+                ShowModal();
+                SetAsMinsize();
+            }
+            else
+            {
+                Hide();
+            }
+
+            UpdateDeleteButton();
+            UpdateMoveButton();
+        }
+    }
+
+    /// <summary>
+    ///   The placed organelle to be shown options of.
+    /// </summary>
+    public OrganelleTemplate SelectedOrganelle
+    {
+        get => selectedOrganelle;
+        set
+        {
+            selectedOrganelle = value;
+            UpdateOrganelleNameLabel();
+        }
+    }
+
+    public bool EnableDeleteOption
+    {
+        get => enableDelete;
+        set
+        {
+            enableDelete = value;
+            UpdateDeleteButton();
+        }
+    }
+
+    public bool EnableMoveOption
+    {
+        get => enableMove;
+        set
+        {
+            enableMove = value;
+            UpdateMoveButton();
+        }
+    }
+
+    public override void _Ready()
+    {
+        selectedOrganelleNameLabel = GetNode<Label>(SelectedOrganelleNameLabelPath);
+        deleteButton = GetNode<Button>(DeleteButtonPath);
+        moveButton = GetNode<Button>(MoveButtonPath);
+
+        UpdateOrganelleNameLabel();
+        UpdateDeleteButton();
+        UpdateMoveButton();
+    }
+
+    private void OnDeletePressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+
+        EmitSignal(nameof(DeletePressed));
+
+        Hide();
+    }
+
+    private void OnMovePressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+
+        EmitSignal(nameof(MovePressed));
+
+        Hide();
+    }
+
+    private void OnModifyPressed()
+    {
+        throw new NotImplementedException();
+    }
+
+    private void UpdateButtonContentsColour(string optionName, bool pressed)
```

Comment:
This doesn't seem to be used from anywhere?
@Kasterisk wasn't it you who wrote this class?

Reply:
That method is connected to the menu items button down/up [signals](https://github.com/Revolutionary-Games/Thrive/blob/14c2eee665296a4a3040ad2b4ac15b729890586f/src/microbe_stage/editor/OrganellePopupMenu.tscn#L373).

Reply:
Ah, okay, that also explains why the parameter needs to be passed a string which is used to construct the path.